tgui/packages/tgui/interfaces/LibraryConsole.js

```diff
@@ -0,0 +1,988 @@
+                value={title}
+                placeholder={title || 'Title'}
+                mt={0.5}
+                onChange={(e, value) => act('set_search_title', {
+                  title: value,
+                })} />
+            </Stack.Item>
+            <Stack.Item>
+              <Input
+                value={author}
+                placeholder={author || 'Author'}
+                mt={0.5}
+                onChange={(e, value) => act('set_search_author', {
+                  author: value,
+                })} />
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+        <Stack.Item>
+          <Button
+            disabled={!can_db_request}
+            textAlign="right"
+            onClick={() => act('search')}
+            color={params_changed ? 'good' : ''}
+            icon="book">
+            Search
+          </Button>
+          <Button
+            disabled={!can_db_request}
+            textAlign="right"
+            onClick={() => act('clear_data')}
+            color="bad"
+            icon="fire">
+            Reset Search
+          </Button>
+        </Stack.Item>
+      </Stack>
+      <Table>
+        <Table.Row>
+          <Table.Cell
+            fontSize={1.5}>
+            #
+          </Table.Cell>
+          <Table.Cell
+            fontSize={1.5}>
+            Category
+          </Table.Cell>
+          <Table.Cell
+            fontSize={1.5}>
+            Title
+          </Table.Cell>
+          <Table.Cell
+            fontSize={1.5}>
+            Author
+          </Table.Cell>
+        </Table.Row>
+        {records.map(record => (
+          <Table.Row key={record.key}>
+            <Table.Cell>
+              <Button
+                onClick={() => act('print_book', {
+                  book_id: record.id,
+                })}
+                icon="print">
+                {record.id}
+              </Button>
+            </Table.Cell>
+            <Table.Cell>
+              {record.category}
+            </Table.Cell>
+            <Table.Cell>
+              {record.title}
+            </Table.Cell>
+            <Table.Cell>
+              {record.author}
+            </Table.Cell>
+          </Table.Row>
+        ))}
+      </Table>
+    </Box>
+  );
+};
+
+export const Upload = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    can_db_request,
+    has_scanner,
+    has_cache,
+    cache_title,
+    cache_author,
+    cache_content,
+  } = data;
+  const [uploadToDB, setUploadToDB] = useLocalState(
+    context, 'UploadDB', false);
+  if (!has_scanner) {
+    return (
+      <NoticeBox>
+        No nearby scanner detected, construct one to continue.
+      </NoticeBox>
+    );
+  }
+  if (!has_cache) {
+    return (
+      <NoticeBox>
+        Scan in a book to upload.
+      </NoticeBox>
+    );
+  }
+  const contentHtml = {
+    __html: run_marked_default(sanitizeText(cache_content)),
+  };
+  return (
+    <Box height="100%">
```

Comment:
Suggesting to replace this with a fragment `<></>`, since modal doesn't need to be contained in a div, and fragments tags conveniently disappear in the final markup.

Reply:
honestly didn't know that was possible. sure